protocols/api/src/main/java/org/apache/james/protocols/api/Encryption.java

```diff
@@ -19,26 +19,41 @@
 
 package org.apache.james.protocols.api;
 
+import java.util.List;
+
 import javax.net.ssl.SSLContext;
 import javax.net.ssl.SSLEngine;
 
 import org.apache.commons.lang3.ArrayUtils;
 
+import io.netty.buffer.ByteBufAllocator;
+import io.netty.handler.ssl.JdkSslContext;
+import io.netty.handler.ssl.SslContext;
+import io.netty.handler.ssl.SslContextBuilder;
+
 /**
  * This class should be used to setup encrypted protocol handling
+ * 
+ * It's recommended to use Netty {@link SslContext} in preference to the JDK's {@link SSLContext}.
+ * Netty {@link SslContext} is much easier to configure, use and supports more features.
+ * 
+ * Netty {@link SslContext} can be configured to use Google's BoringSSL.
+ * BoringSSL is a clone of OpenSSL with all redundant/old ciphers removed and is compiled into to all the Chrome/Chromium browsers.
+ * Simply include the 'io.netty:netty-tcnative-boringssl-static' dependency as discussed here: https://netty.io/wiki/forked-tomcat-native.html
+ * You should regularly update the version of this library to keep in-sync with the latest version used by Chrome/Chromium browsers.
+ * 
+ * Use {@link SslContextBuilder} to create an instance of {@link SslContext}.
  */
-public final class Encryption {
+public abstract class Encryption {
```

Comment:
I worked on a clean abstraction, turning Encryption into an interface, moving it to protocols-netty, and providing Netty implementation of it.

Though I failed at using Netty native bindings... netty-ctnative was built against openssl-1.0 while I have openssl-1.1... Feeling sad here!

I'll PR this generalisation of OpenSSL separately...